pyprof/prof/convert.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, d):
 
         assert (mod == "Tensor")
         assert (op in Convert.ops)
-        assert (len(args) == 1)
+        assert (len(args) >= 1), "Args {} Len {}".format(args, len(args))
```

Comment:
Can we move this to a separate PR. Was there a reason to weaken the assertion? If the assertion indeed got triggered, we should fix it as a separate PR.

Reply:
This assertion was changed because the filtering of args is broken.

It appears that the software is trying to somehow keep track of keyword args and args, but it is not at all clear from any of the code that this is the case.

I think the code is assuming that args with the name key == "" means that it was not a keyword argument.  And args that have the name field set are keyword arguments.  For some reason that is also not clear from any of the code, the assumption is that unnamed args refer to the tensors and named args refer to hyper parameters.

I want the unique name of the ops to be passed down from nvtx markers so we can use them to identify the ops in the post processing stage.

Reply:
That is the Python definition of `args` and `kwargs`. `args` are passed as a list (without names) and `kwargs` are passed as a dictionary (key-value pair). This fact was used when the NVTX marker was created. `name == ""` when its args and `name == foo` when its kwargs.

`args` and `kwargs` has no relationship to Pytorch / tensors / hyperparameters. It just depends on how the user decided to pass an argument to a function.

If you want unique name for ops, we can create an extra field in the NVTX marker. It should be really simple to do and does not require any hacking / overloading of existing keys in the marker.